include/engine/datafacade/shared_datafacade.hpp

```diff
@@ -68,6 +69,7 @@ class SharedDataFacade final : public BaseDataFacade
     std::unique_ptr<storage::SharedMemory> m_layout_memory;
     std::unique_ptr<storage::SharedMemory> m_large_memory;
     std::string m_timestamp;
+    extractor::ProfileProperties* m_profile_properties;
```

Comment:
Ownership is clearly managed actually. None of the types here have ownership of their containing data. Its all managed by the shared memory segment. We could copy, but that would be inconsistent with everything else.